weaver/main.go

```diff
@@ -143,7 +144,9 @@ func main() {
 	}
 
 	router.Start()
-	initiateConnections(router, peers)
+	if errors := router.ConnectionMaker.InitiateConnections(peers); len(errors) > 0 {
+		log.Fatal(strings.Join(errorMessages(errors), ", "))
```

Comment:
That looks gross in the logs. Make it a "\n", which also means the `join` can be handled by `errorMessages`, since "\n" is what gets used at the other call site